cpp/tests/utilities/cudf_gtest.hpp

```diff
@@ -92,3 +92,24 @@ struct TypeList<Types<TYPES...>> {
 }  // namespace testing
 
 #include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+// Utility for testing the expectation that an expression x throws the specified
+// exception whose what() message ends with the msg
+#define EXPECT_THROW_MESSAGE(x, exception, startswith, endswith)     \
```

Comment:
It's good to test that an exception is thrown in exceptional cases, but I think testing for an exact error message is a bit too much on the [white-box](https://www.guru99.com/white-box-testing.html) end of the spectrum.

Reply:
You don't have to remember, you just have to run tests.

Reply:
An alternative is declaring unique const error codes (integer) and expecting those. Could be static on the errors themselves,
```
cudf::specific_logic_error::error_id
```
or injectable
```
cudf::logic_error("message", cudf::error_codes::silly_mistake)
```

Reply:
> You don't have to remember, you just have to run tests.

Is it really important to test the exact contents of the message?

Reply:
There's no other way to determine that an error was thrown for the reason you expect.

Reply:
Removed. :(